Database.py

```diff
@@ -32,58 +43,106 @@ def get_stops_from_function(self,function:str)->list:
 
         return self.cursor.execute(query).fetchall()
 
-    def add_lines_to_json(self,dir):
+    def add_lines_to_json(self, dir: str):
+        """
+        Adds lines to a JSON database. Line = {"name", "stops", "times"}.
+        """
+        used_stops = set()  # used for checking if all stops from the database were used in the process of extraction from html templates
+        lines1 = []
 
-        lines1 =[]
         for files in os.listdir(dir):
-            file = open(f"D:\PyCharm\PyCharm 2023.2.4\JakNieDojade\Autobusy\\{files}","r",encoding="UTF-8")
+            path = os.path.join(dir, files)
+            file = open(path, "r", encoding="UTF-8")
 
             name = file.name
             line = []
             stops = []
             times = []
             cur = 0
 
-            for lines in file:
-                if 'Przystanek' in lines and "onlyPrint" in lines:
+            for lines in file:  # loop for selection stop names, these names in Wroclaw's MPK webpage are stored in lines where "Przystanek" and "onlyPrint" occur, times are stored with "'" at the end of a line
+                if 'Przystanek ' in lines and "onlyPrint" in lines:
                     stops.append(lines[lines.rfind(">") + 2:-1])
+
                 if "Czas przejazdu" in lines and "'" in lines:
                     times.append(int(lines[lines.rfind(">") + 2:-2]) - cur)
                     cur = int(lines[lines.rfind(">") + 2:-2])
 
-            res = []
+            res = []  # stores tuples with the ID and name of a stop
+            stop_names = []
+
             for stop in stops:
-                if 'ROD' in stop:
+                if 'ROD' in stop:  # check for mismatches in the names from the file and the database
                     stop = stop.title()
-                temp = self.cursor.execute(f"SELECT IdP,Nazwa FROM Przystanki WHERE Nazwa = '{stop}'").fetchone()
+
+                if "P&amp;R" in stop:
+                    stop = stop.replace("P&amp;R", "PR")
+
+                temp = self.cursor.execute(
+                    f"SELECT IdP,Nazwa FROM Przystanki WHERE Nazwa = '{stop}'").fetchone()  # select ID of the stop
+                used_stops.add(temp[0])
                 res.append(temp[0])
+                stop_names.append(temp[1])
+
+            logger.info(
+                f"added {name[-13:]},first stop {stop_names[0]} kod {res[0]}, last stop {stop_names[-1]} kod {res[-1]} \nPrzystanki: {stop_names}, Czasy {times}\n Dlugosc przystankow po nazwach {len(stop_names)}, po kodach {len(res)}, dlugosc czasow {len(times)}")  # logging for debugging purposes
+
+            if len(res) - 1 != len(times):
+                logger.error(
+                    f"Line {name[-13:]} lenghts dont align")  # throw an error in the log to signal that the webpage did not contain all necessary info, important for further graphing
```

Comment:
nie wszystkie 'name' mają 13 znaków, np. " Linia_133_3.txt"; 
name[-13:] wyprintuje ci: "nia_133_3.txt". Wsm korekta raczej nie wymagana, bo obiektywnie nie ma to znaczenia, jako że jest to tylko for debuggin purposes